### ISSUE TITLE
turn gl.h into an umbrella header

### ISSUE THREAD

**Issue by richardeakin** [cinder/Cinder#600]
I've been wondering if we couldn't restructure the gl headers a bit to make our day to day lives easier in terms of having to #include things and what not.  This is a bigger deal in glNext because the number of headers in cinder/gl/* has grown from 12 to 21, and will likely grow a bit more before it is finished.

So, strawman proposal for gl headers:

* the current contents of gl.h is separated into components:
  * cinder/gl/platform.h: all of the glload and platform specific headers. Basically all the cross platform preprocessor craziness
  * cinder/gl/draw.h: convenience drawing utils
  * cinder/gl/scoped.h: all the RAII binding stuff
  * ~~cinder/gl/utils.h~~ cinder/gl/wrapper.h: gl functionality wrapped in namespace `ci::gl`
    * After beginning on the re-organization, a utils.h file doesn't make sense for things like 
* then, gl.h becomes an umbrella header: is made to include all of these, and also all of the other widely used gl headers
  * gl/Texture.h, gl/Fbo.h, gl/Shader.h, gl/GlslProg.h, etc..

*Note*: I've chosen to use lower case filenames as there are no public classes in these files. If others would instead prefer them to star  upper case than speak now. :)

This approach would be backwards compatible, but also allow people to include what they needed if thats how they want to do it.

One thing that got me thinking about this (again) was reading about [clang's experimental module support](http://clang.llvm.org/docs/Modules.html), and how such a structure would facilitate that eventually, if that feature makes it into the C++ standard.

**Comment by richardeakin** [comment]
I've begun this work, and immediately I noticed that some of the functionality in the existing gl.h doesn't fit into any of the above categories, which are the C++ gl wrapper methods, ex. `gl::vertex()`, `gl::polygonMode()`, `gl::enableVertexAttribArray()`, etc.  So, I think a more suitable name would be cinder/gl/wrapper.h. I'm inclined to put any other stuff that would have gone into utils.h in wrapper.h instead, since there isn't much.

Issue description has been updated to reflect this.
